api/v1alpha1/aws_clusterconfig_types.go

```diff
@@ -22,7 +20,6 @@ func (AWSSpec) VariableSchema() clusterv1.VariableSchema {
 			Properties: map[string]clusterv1.JSONSchemaProps{
 				"region": Region("").VariableSchema().OpenAPIV3Schema,
 			},
-			Required: []string{"region"},
```

Comment:
Yes its required, but even there it can be set either in the API or come from the credentials config or env variable on the controller.

We do get a clear error on the `AWSCluster` though. 
```
    - lastTransitionTime: "2023-10-11T14:07:35Z"
      message: 'failed to create new vpc: failed to create vpc: MissingRegion: could
        not find region configuration'
      reason: VpcReconciliationFailed
      severity: Warning
      status: "False"
      type: VpcReady
```

Reply:
But how do I inform a user that it is indeed required? Perhaps we could do that with settings on the hook registration in the CC? I'm thinking how a UI could generate forms and have required fields actually annotated as such

Reply:
Talked offline, we'll keep this project minimally opinionated and instead provide a mechanism for consumers to further configure required variables and their defaults.